app.rb

```diff
@@ -0,0 +1,164 @@
+    input = gets.chomp
+    case input.to_i
+    when 1
+      student_option
+    when 2
+      teacher_option
+    else
+      puts 'you entered invalid input! Please try again:'
+      return
+    end
+    puts 'Person selected succesfully'
+  end
+
+  def permission?(parent_permission)
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+    case permission
+    when 'n', 'N'
+      !parent_permission
+    when 'y', 'Y'
+      parent_permission
+    else
+      permission?(parent_permission)
+    end
+  end
+
+  def create_student(classroom, age, name, parent_permission)
+    student = Student.new(classroom, age, name, parent_permission: parent_permission)
+    @people << student unless @people.include?(student)
+    @students << student unless @students.include?(student)
+  end
+
+  def create_teacher(specialization, age, name)
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher unless @people.include?(teacher)
+    @teachers << teacher unless @teachers.include?(teacher)
+  end
+
+  def student_option
+    print ' Enter student Name : '
+    name = gets.chomp
+    print ' Enter student Age : '
+    age = gets.chomp
+    print ' Enter student Classroom <number> : '
+    classroom = gets.chomp
+    parent_permission = true
+    perm = permission?(parent_permission)
+    create_student(classroom.to_i, age.to_i, name, perm)
+    puts "---------  New student #{name} has been Added successfully!  ----------- \n
+  #{name} is #{age} years old is in classroom #{classroom.to_i}"
+    puts '-------------------------------- '
+  end
+
+  def teacher_option
+    print 'Enter teacher name:'
+    name = gets.chomp
+    print 'Enter teacher age:'
+    age = gets.chomp
+    print 'Enter Teacher specialization:'
+    specialization = gets.chomp
+    create_teacher(specialization, age.to_i, name)
+    puts '-----New Teacher Added Successfully----'
+    puts "#{name} is #{age} years old, specialization is #{specialization}"
+    puts '-------------------------'
+  end
+
+  def create_new_book
+    puts '------------ Creating New Book -----------'
+    print ' Enter Book Title : '
+    title = gets.chomp
+    print ' Enter Book Author : '
+    author = gets.chomp
+
+    book = Book.new(title, author)
+    @books << book unless @books.include?(book)
+    puts '------------ New Book Added -----------'
+    puts "#{title} by #{author} added to book list successfuly!\n
+  -----------------------------------"
+  end
+
+  def create_rental
+    puts '------------ Making a New Rental -----------'
+    puts ' Select a Book from the list below by the number'
+    list_books
+    print ' Enter book number : '
+    book_number = gets.chomp
+    puts 'Select a Person from the list below by the number'
+    list_people
+    person_number = gets.chomp
+    print ' Enter the Date e.g (2022/9/29) : '
+    date = gets.chomp
+
+    rent = Rental.new(date, books[book_number.to_i - 1], people[person_number.to_i - 1])
```

Comment:
- [ ] Your app is throwing an error in an attempt to create a rental. An error is thrown just after entering details to create a rental. Kindly resolve the same.

**Creating rental error:**
![image](https://user-images.githubusercontent.com/12745474/193356526-270ab870-745a-450d-b96f-969eb64700f8.png)